functions/api/transactions.js

```diff
@@ -76,19 +103,93 @@ router.get("/:id", isAdmin, async (request, response, next) => {
         transactionCollection = transactionCollection.where(constants.DATE,">=",new Date(fromDate));
     }
     if(toDate){
-        transactionCollection = transactionCollection.where(constants.DATE,"<=",new Date(toDate));
+        toDate = new Date(toDate)
+        // adding a day to the given query date as the new Date function returns 00:00:00 and we 
+        // need to include the toDate for query .
+        toDate.setDate(toDate.getDate()+1)
+        console.log(toDate);
+        transactionCollection = transactionCollection.where(constants.DATE,"<",toDate);
+    }
+    if(nextPageToken !== null && prevPageToken !== null) {
+        response.status(400)
+        .send("Invalid Query Parameters, Please send either nextPageToken or prevPageToken");
+
+    }
+    var transCollectionOrig = transactionCollection
+    const res = {}
+    if(nextPageToken === null && prevPageToken === null) {
+            transactionCollection = transactionCollection
+                .limit(pageSize)
+            //res[constants.PREV_PAGE_TOKEN]=null
+    }
+    if(nextPageToken !== null ) {
+        var lastVisibleDoc = await db.collection(constants.BRANCHES)
+            .doc(branchId)
+            .collection(constants.TRANSACTIONS)
+            .doc(nextPageToken)
+            .get()
+        console.log (lastVisibleDoc.id);
+        if(lastVisibleDoc){
+            transactionCollection = transactionCollection
+                .startAfter(lastVisibleDoc)
+                .limit(pageSize)
+            }
+    }
+    if(prevPageToken !== null) {
+    var prevVisibleDoc = await db.collection(constants.BRANCHES)
+            .doc(branchId)
+            .collection(constants.TRANSACTIONS)
+            .doc(prevPageToken)
+            .get()
+        console.log (prevVisibleDoc.id);
+        if(prevVisibleDoc){
+            transactionCollection = transactionCollection
+                .endBefore(prevVisibleDoc)
+                .limitToLast(pageSize)
+        }
     }
     let snapshot = await transactionCollection.get()
+    //To identify if its end of pages ( no prev or no next)
+    hasPreviousPage(transCollectionOrig,snapshot)
+        .then(hasPrevious => console.log(`in then loop : ${hasPrevious}`))
```

Comment:
does hasPreviewPage promise resolve anything. whats the value of **hasPrevious**

Reply:
@neerk28  actually am intended to return a boolean true or false from the function hasPreviousPage. but when i just do a simple return its returning some promise object.. so i was reading thru that part , keeping this pr open.. as i thought i remvoed these changes..

Reply:
can u suggest on how to resolve it ..

Reply:
if you do above changes in hasPreviousPage; then "hasPrevious" will be true/false here.

Reply:
fixed with latest commits